### PR TITLE
Breakpoints removed from the sidebar on notebook closed

### PR DESCRIPTION
**Pull request by KsavinN**
Resolved by Jeremy in PR #308 

**This branch needs to be rebased and refactored after this PR #290** 

resolve issue #168.

- [x] after close all notebooks clear breakpoints.

After close widget:

clearBreakpoints

### CODE REVIEW

**Comment by jtpio** [comment]
Should we really clear the breakpoints from the state in the kernel? Or just from the UI?

If we re-open the notebook the breakpoints will be gone, even if the kernel was not shut down.

**Comment by KsavinN** [comment]
> Should we really clear the breakpoints from the state in the kernel? Or just from the UI?
> 
> If we re-open the notebook the breakpoints will be gone, even if the kernel was not shut down.

It should still have breakpoints.  It should only clear UI.
stillSHow

File: src/index.ts [hunk 1]
```diff
@@ -125,6 +125,8 @@ async function updateToolbar(
           debug.session.debuggedClients.add(debug.session.client.path);
           await debug.start();
         } else {
+          await debug.restart();
+          await debug.clearBreakpoints();
```
Comment (jtpio) [inline src/index.ts]: Not sure we should clean the breakpoints separately, as this looks like a special case.

**Comment by jtpio** [comment]
This shouldn't be specific to breakpoints only. There is a similar issue when the execution is stopped and the notebook is closed:

dispose-on-execution-stopped

Maybe there could be a more generic model cleanup method that would be triggered when the widget is disposed? The sidebar would react to model changes and clean itself up.

**Comment by jtpio** [comment]
Ideally this would also handle kernel changes, so the sidebar UI would be cleared when the kernel is changed:

clear-sidebar-kernel-changed

**Pull request closed without merging**